stripecardscan/src/main/java/com/stripe/android/stripecardscan/cardscan/CardScanActivity.kt

```diff
@@ -0,0 +1,683 @@
+//            instanceId = Stats.instanceId,
+//            scanId = Stats.scanId,
+//            device = Device.fromContext(this),
+//            appDetails = AppDetails.fromContext(this),
+//            scanStatistics = ScanStatistics.fromStats()
+//        )
+        setFlashlightState(false)
+        finish()
+    }
+
+    private fun displayState(newState: ScanState, previousState: ScanState?) {
+        when (newState) {
+            is ScanState.NotFound -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeNotFoundBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_not_found)
+                viewBinding.viewFinderBorder.startAnimation(R.drawable.stripe_card_border_not_found)
+                viewBinding.instructions.setText(R.string.stripe_card_scan_instructions)
+            }
+            is ScanState.FoundShort -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeFoundBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_found)
+                viewBinding.viewFinderBorder.startAnimation(R.drawable.stripe_card_border_found)
+                viewBinding.instructions.setText(R.string.stripe_card_scan_instructions)
+                viewBinding.instructions.show()
+            }
+            is ScanState.FoundLong -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeFoundBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_found)
+                viewBinding.viewFinderBorder
+                    .startAnimation(R.drawable.stripe_card_border_found_long)
+                viewBinding.instructions.setText(R.string.stripe_card_scan_instructions)
+                viewBinding.instructions.show()
+            }
+            is ScanState.Correct -> {
+                viewBinding.viewFinderBackground
+                    .setBackgroundColor(getColorByRes(R.color.stripeCorrectBackground))
+                viewBinding.viewFinderWindow
+                    .setBackgroundResource(R.drawable.stripe_card_background_correct)
+                viewBinding.viewFinderBorder.startAnimation(R.drawable.stripe_card_border_correct)
+                viewBinding.instructions.hide()
+            }
+        }
+    }
+
+    private fun changeScanState(newState: ScanState): Boolean {
+        if (newState == scanStatePrevious || scanStatePrevious?.isFinal == true) {
+            return false
+        }
+
+        scanState = newState
+        displayState(newState, scanStatePrevious)
+        scanStatePrevious = newState
+        return true
+    }
```

Comment:
I wonder if it would be possible to use generics to move this method and `displayState` into a separate super class, something like `SimpleScanStatefulActivity`. What are your thoughts?